domain/agentbinary/state/types.go

```diff
@@ -21,3 +23,26 @@ type agentBinaryRecord struct {
 	ArchitectureID  int    `db:"architecture_id"`
 	ObjectStoreUUID string `db:"object_store_uuid"`
 }
+
+type metadataRecord struct {
```

Comment:
We need to add architecture to the records returned. See https://github.com/juju/juju/blob/37616f315a9c560cd19ed757de0d50c0c1e817eb/apiserver/common/tools.go#L369 we do use it through version.Binary.

Reply:
You are right, thanks!